code/modules/wiremod/components/atom/remotecam.dm

```diff
@@ -0,0 +1,421 @@
+	else
+		. += create_ui_notice("Energy Usage While Active: [display_energy(current_camera_range > 0 ? REMOTECAM_ENERGY_USAGE_FAR : REMOTECAM_ENERGY_USAGE_NEAR)] Per [DisplayTimeText(COMP_CLOCK_DELAY)]", "orange", "clock")
+
+/obj/item/circuit_component/remotecam/populate_ports()
+	start = add_input_port("Start", PORT_TYPE_SIGNAL)
+	stop = add_input_port("Stop", PORT_TYPE_SIGNAL)
+	if(camera_range_settable)
+		camera_range = add_input_port("Camera Range", PORT_TYPE_NUMBER, default = 0)
+	network = add_input_port("Network", PORT_TYPE_STRING, default = "ss13")
+
+	if(camera_range_settable)
+		current_camera_range = camera_range.value
+	c_tag_random = rand(1, 999)
+
+/obj/item/circuit_component/remotecam/register_shell(atom/movable/shell)
+	shell_parent = shell
+	stop_process()
+
+/obj/item/circuit_component/remotecam/unregister_shell(atom/movable/shell)
+	stop_process()
+	remove_camera()
+	shell_parent = null
+
+/obj/item/circuit_component/remotecam/Destroy()
+	stop_process()
+	remove_camera()
+	shell_parent = null
+	return ..()
+
+/obj/item/circuit_component/remotecam/input_received(datum/port/input/port)
+	if(shell_parent && shell_camera)
+		update_camera_name_network()
+		if(COMPONENT_TRIGGERED_BY(start, port))
+			start_process()
+			cameranet_add()
+			current_camera_state = TRUE
+		else if(COMPONENT_TRIGGERED_BY(stop, port))
+			stop_process()
+			close_camera() //Instantly turn off the camera
+			current_camera_state = FALSE
+
+/**
+ * Initializes the camera
+ */
+/obj/item/circuit_component/remotecam/proc/init_camera()
+	shell_camera.desc = "This camera belongs in a circuit. If you see this, tell a coder!"
+	shell_camera.AddElement(/datum/element/empprotection, EMP_PROTECT_ALL)
+	shell_camera.use_power = NO_POWER_USE
+	shell_camera.start_active = TRUE
+	shell_camera.internal_light = FALSE
+	current_camera_name = ""
+	if(camera_range_settable)
+		current_camera_range = camera_range.value
+	current_cameranet_state = TRUE
+	current_camera_emp = FALSE
+	current_camera_network = ""
+	close_camera()
+	update_camera_range()
+	update_camera_name_network()
+	if(current_camera_state)
+		start_process()
+		update_camera_location()
+	else
+		cameranet_remove() //Remove camera from global cameranet until user activates the camera first
+	if(!camera_signal_move_override)
+		RegisterSignal(shell_parent, COMSIG_MOVABLE_MOVED, PROC_REF(update_camera_location))
+	RegisterSignal(shell_parent, COMSIG_ATOM_EMP_ACT, PROC_REF(set_camera_emp))
+
+/**
+ * Remove the camera
+ */
+/obj/item/circuit_component/remotecam/proc/remove_camera()
+	if(shell_camera)
+		if(!camera_signal_move_override)
+			UnregisterSignal(shell_parent, COMSIG_MOVABLE_MOVED)
+		UnregisterSignal(shell_parent, COMSIG_ATOM_EMP_ACT)
+		if(current_camera_emp)
+			deltimer(current_camera_emp_timer_id)
+			current_camera_emp = FALSE
+		cameranet_add() //Readd camera to cameranet before deleting camera
+		QDEL_NULL(shell_camera)
```

Comment:
You'll really want to go for early returns here.

I.e.
```dm
if(!shell_camera)
  return
```
It makes the code a lot more readable